cardano-api/src/Cardano/Api/IPC.hs

```diff
@@ -166,9 +166,10 @@ data LocalNodeConnectInfo mode =
 
 type MinNodeToClientVersion = NodeToClientVersion
 
-data QueryError
+data QueryError a
   = QueryErrorAcquireFailure !Net.Query.AcquireFailure
   | QueryErrorUnsupportedVersion !MinNodeToClientVersion !NodeToClientVersion
+  | QueryErrorOf a
```

Comment:
I'm not sure I see the benefit of parameterizing `QueryError`. Was the `QueryErrorOf` constructor defined to indicate that a particular error is a query error?